tests/tools/test_dist_part.py

```diff
@@ -0,0 +1,226 @@
+            assert feat_array.shape[0] == num_edges[etype] // num_chunks
+
+def test_partition():
+    # Step1: graph partition
+    output_dir = '2parts'
+    os.system('python tools/partition_algorithms/random_partition.py '\
+              '--metadata chunked-data/metadata.json --output_path {} --num_partitions 2'.format(output_dir))
+    for ntype in ['author', 'institution', 'paper']:
+        fname = os.path.join(output_dir, '{}.txt'.format(ntype))
+        with open(fname, 'r') as f:
+            header = f.readline().rstrip()
+            assert isinstance(int(header), int)
+
+def test_dispatch():
+    # Step2: data dispatch
+    out_dir = 'partitioned'
+    num_parts = 2
+    with open('ip_config.txt', 'w') as f:
+        f.write('127.0.0.1\n')
+        f.write('127.0.0.2\n')
+
+    os.system('python tools/dispatch_data.py --in-dir chunked-data '\
+              '--partitions-dir 2parts --out-dir {} --ip-config ip_config.txt'.format(out_dir))
+
+    # check metadata.json
+    meta_fname = os.path.join(out_dir, 'metadata.json')
+    with open(meta_fname, 'rb') as f:
+        meta_data = json.load(f)
+
+    all_etypes = ['affiliated_with', 'writes', 'cites', 'rev_writes']
+    for etype in all_etypes:
+        assert len(meta_data['edge_map'][etype]) == num_parts
+    assert meta_data['etypes'].keys() == set(all_etypes)
+    assert meta_data['graph_name'] == 'mag240m'
+
+    all_ntypes = ['author', 'institution', 'paper']
+    for ntype in all_ntypes:
+        assert len(meta_data['node_map'][ntype]) == num_parts
+    assert meta_data['ntypes'].keys() == set(all_ntypes)
+    assert meta_data['num_edges'] == 4200
+    assert meta_data['num_nodes'] == 720
+    assert meta_data['num_parts'] == num_parts
+    assert meta_data['part_method'] == 'metis'
+
+    for i in range(num_parts):
+        sub_dir = 'part-' + str(i)
+        assert meta_data[sub_dir]['node_feats'] == 'part{}/node_feat.dgl'.format(i)
+        assert meta_data[sub_dir]['edge_feats'] == 'part{}/edge_feat.dgl'.format(i)
+        assert meta_data[sub_dir]['part_graph'] == 'part{}/graph.dgl'.format(i)
+
+        # check data
+        sub_dir = os.path.join(out_dir, 'part' + str(i))
+
+        # graph.dgl
+        fname = os.path.join(sub_dir, 'graph.dgl')
+        assert os.path.isfile(fname)
+        g_list, data_dict = load_graphs(fname)
+        g = g_list[0]
+        assert isinstance(g, dgl.DGLGraph)
+
+        # node_feat.dgl
+        fname = os.path.join(sub_dir, 'node_feat.dgl')
+        assert os.path.isfile(fname)
+        tensor_dict = load_tensors(fname)
+        all_tensors = ['paper/feat', 'paper/label', 'paper/year']
+        assert tensor_dict.keys() == set(all_tensors)
+        for key in all_tensors:
+            assert isinstance(tensor_dict[key], torch.Tensor)
+
+        # edge_feat.dgl
+        fname = os.path.join(sub_dir, 'edge_feat.dgl')
+        assert os.path.isfile(fname)
+        tensor_dict = load_tensors(fname)
+
+if __name__ == '__main__':
```

Comment:
are the generated files/dirs removed/deleted once test finishes?

Reply:
Jinjing told me before that all the generated files will be removed after the CI is finished.

Reply:
this is true as we're running in docker which will be destroyed once finished. But I think it's a better practice to always remove tmp dir/files once specific tests finishes. If we have many tests and they all rely on the delete logic when container destroys, then the total occupied disk may be not trivial. It gets worse when multiple docker are running at the same time.

Reply:
Ok. I will remove them then.

Reply:
Perhaps use python https://docs.python.org/3/library/tempfile.html

Reply:
Pytest also provides default way to request temp file: https://docs.pytest.org/en/7.1.x/getting-started.html#request-a-unique-temporary-directory-for-functional-tests

Reply:
I've used `tempfile.TemporaryDirectory`.